apps/tlon-mobile/ios/Shared/Notifications/PushNotificationManager.swift

```diff
@@ -224,3 +64,71 @@ enum NotificationCategory: String {
         }
     }
 }
+
+/**
+ This type's value can be represented as a user-facing push alert.
+ */
+protocol UNNotificationRenderable {
+  /**
+   Renders the receiver onto the specified notification content. Caller should discard the input
+   `UNMutableNotificationContent` and use the returned value, since the identity of the notification
+   content may change.
+   
+   ```swift
+   let (content, msgIntent) = await payload.render(to: mutableContent)
+   // `mutableContent` should no longer be used; use `content` instead
+   if let msgIntent {
+     // donate intent here
+   }
+   ```
+   */
+  func render(to content: UNMutableNotificationContent) async -> (UNNotificationContent, INSendMessageIntent?)
+}
+
+extension Yarn: UNNotificationRenderable {
```

Comment:
sorry about code movement here – I made this change when I was implementing activity event stuff so I could just adopt `UNNotificationRenderable` on `ActivityEvent`, then I backed out of that – figured it's a fine change to keep